guardian/typerighter
# Resolve critical vulnerability in helper script

## What does this change?

Switch from libxml2js to libxml2-wasm. libxml2js is no longer actively maintained and has a critical vulnerability.

Also adds some more type information and a script to check that the types are valid.

## How to test

`yarn check-types` to show that Typescript is happy

`node compare-rule-xml.js fixtures/example-rules-old.xml fixtures/example-rules-new.xml fixtures/example-rule-ids.txt` to see that script still runs.
